panel/template/fast/settings/fast_template_settings.py

```diff
@@ -0,0 +1,347 @@
+""" # noqa: E501
+
+GEAR_SVG = """
+<svg xmlns="http://www.w3.org/2000/svg" width="1.3em" style="display: inline-block;vertical-align: middle;" fill="currentColor" class="bi bi-gear" viewBox="0 0 16 16">
+  <path d="M8 4.754a3.246 3.246 0 1 0 0 6.492 3.246 3.246 0 0 0 0-6.492zM5.754 8a2.246 2.246 0 1 1 4.492 0 2.246 2.246 0 0 1-4.492 0z"/>
+  <path d="M9.796 1.343c-.527-1.79-3.065-1.79-3.592 0l-.094.319a.873.873 0 0 1-1.255.52l-.292-.16c-1.64-.892-3.433.902-2.54 2.541l.159.292a.873.873 0 0 1-.52 1.255l-.319.094c-1.79.527-1.79 3.065 0 3.592l.319.094a.873.873 0 0 1 .52 1.255l-.16.292c-.892 1.64.901 3.434 2.541 2.54l.292-.159a.873.873 0 0 1 1.255.52l.094.319c.527 1.79 3.065 1.79 3.592 0l.094-.319a.873.873 0 0 1 1.255-.52l.292.16c1.64.893 3.434-.902 2.54-2.541l-.159-.292a.873.873 0 0 1 .52-1.255l.319-.094c1.79-.527 1.79-3.065 0-3.592l-.319-.094a.873.873 0 0 1-.52-1.255l.16-.292c.893-1.64-.902-3.433-2.541-2.54l-.292.159a.873.873 0 0 1-1.255-.52l-.094-.319zm-2.633.283c.246-.835 1.428-.835 1.674 0l.094.319a1.873 1.873 0 0 0 2.693 1.115l.291-.16c.764-.415 1.6.42 1.184 1.185l-.159.292a1.873 1.873 0 0 0 1.116 2.692l.318.094c.835.246.835 1.428 0 1.674l-.319.094a1.873 1.873 0 0 0-1.115 2.693l.16.291c.415.764-.42 1.6-1.185 1.184l-.291-.159a1.873 1.873 0 0 0-2.693 1.116l-.094.318c-.246.835-1.428.835-1.674 0l-.094-.319a1.873 1.873 0 0 0-2.692-1.115l-.292.16c-.764.415-1.6-.42-1.184-1.185l.159-.291A1.873 1.873 0 0 0 1.945 8.93l-.319-.094c-.835-.246-.835-1.428 0-1.674l.319-.094A1.873 1.873 0 0 0 3.06 4.377l-.16-.292c-.415-.764.42-1.6 1.185-1.184l.292.159a1.873 1.873 0 0 0 2.692-1.115l.094-.319z"/>
+</svg>
+""" # noqa: E501
+
+
+class FastTemplateSettings(Viewer):
+    """
+    The FastTemplateSettings component enables you to change the look
+    and feel of the FastListTemplate and FastGridTemplate dynamically
+    on a live app.
+
+    Add it to the sidebar of your template to dynamically change the
+    settings of the Template.
+
+    If you need more control you can add the `comms` and `editor`
+    components individually instead.
+    """
+
+    comms = param.ClassSelector(class_=Column, constant=True, doc="""
+        The component that provides bidirectional communication to set
+        and get the style values. Include this in your app if you want
+        to enable changing Template settings dynamically.""")
```

Comment:
Can you explain this to me? This seems like a pretty big hack and makes it difficult for me to decide to merge this.

Reply:
I guess the idea is to include the designers in the layout since they need to be added somewhere. Generally I really dislike this and instead these objects should be inserted as roots on the document so that they are entirely invisible to a user. I can do that refactoring but probably not before a 0.12 release.

Reply:
I was in doubt on whether to always include them or not. My guess is that 

- setting template colors etc. dynamically in a live app will not be used often. And then they will just add extra complexity/ slow down the template. The implication is to not include them in the template.
- users would like this to be as easy to use as possible. The implication is to always include them in the template.

So what to do?

Reply:
>I was in doubt on whether to always include them or not. My guess is that

I'm not suggesting that they should always be added just that the mechanism by which they are added is not what I'd want to see. Ideally users would toggle an option on the template which then ensures these objects are added as roots for the template.